Lib/doctest.py

```diff
@@ -1140,7 +1140,9 @@ def _find_lineno(self, obj, source_lines):
             obj = obj.fget
         if inspect.isfunction(obj) and getattr(obj, '__doc__', None):
             # We don't use `docstring` var here, because `obj` can be changed.
-            obj = inspect.unwrap(obj).__code__
+            obj = getattr(inspect.unwrap(obj), '__code__', None)
+            if obj is None:
+                return None  # if there's no code, there's no lineno
```

Comment:
I would _probably_ do this like this? But just personal preference really; your way is fine!

```suggestion
            obj = inspect.unwrap(obj)
            try:
                obj = obj.__code__
            except AttributeError:
                # Functions implemented in C don't necessarily
                # have a __code__ attribute.
                # If there's no code, there's no lineno
                return None
```

Reply:
Ok, seems cleaner 👍